video-app/src/app/store/videos/video.effects.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+import { YoutubeService } from '@core/youtube/youtube.service';
+import { VimeoService } from '@core/vimeo/vimeo.service';
+
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import * as VideoActions from '@store/videos/video.actions';
+import { Observable, of } from 'rxjs';
+import { Action } from '@ngrx/store';
+import { mergeMap, map, catchError } from 'rxjs/operators';
+
+@Injectable()
+export class VideoEffects {
+  constructor(
+    private readonly youTubeService: YoutubeService,
+    private readonly vimeoService: VimeoService,
+    private readonly action$: Actions
+  ) {}
+
+  AddYouTubeVideo$: Observable<Action> = createEffect(() =>
+    this.action$.pipe(
+      ofType(VideoActions.addYouTubeVideo),
+      mergeMap((action) =>
+        this.youTubeService.addVideo(action.videoId).pipe(
+          map((response) => {
+            const video = this.youTubeService.parseVideo(response, action.videoId);
+            if (video === null) {
+              return VideoActions.addYouTubeVideoFail();
+            } else {

```

Comment:
ten else jest zbedny, wyżej robisz return w ifie więc jak warunek zostanie spełniony to do tego momentu już nie dojdzie

Reply:
ten else jest potrzebny do dodawania video